### PR TITLE
[extraction] extend bigint extraction with division and more comparisons

### PR DESCRIPTION
**Pull request by Lysxia**
Extend #8252 

- [x] Added **changelog**.

### CODE REVIEW

File: doc/changelog/10-standard-library/15098-extract-bigint.rst [hunk 1]
```diff
@@ -0,0 +1,4 @@
+- **Added:**
+  Added more extraction definitions for division and comparison of Z and N
```
Comment (Alizter) [inline doc/changelog/10-standard-library/15098-extract-bigint.rst:2]:
```suggestion
  More extraction definitions for division and comparison of Z and N
```

**Comment by Alizter** [comment]
Can we get some tests?

**Comment by Lysxia** [comment]
Sure, I can add some tests to this PR.

**Comment by Lysxia** [comment]
Added some `TestCompile` tests. Is there also a way to run the compiled tests to make sure the results are as expected?

**Comment by Alizter** [comment]
@Lysxia Yes, look at the output folder in the test-suite. See the documentation here https://github.com/coq/coq/blob/master/test-suite/README.md. If you make the test-suite targeting `approve-output` the output tests which fail, will get their output files updated.

**Comment by Lysxia** [comment]
The `output` folder seems to only test the output of `.v` files, so I could check that extraction produces a certain OCaml program, but that doesn't check the result of interpreting that program, does it?

**Comment by Alizter** [comment]
It doesn't seem possible to add output tests for extracted code currently. We discussed this in last week's coq call. https://github.com/coq/coq/wiki/Coq-Call-2021-12-01

For now I would like to merge this as-is. Are you happy with that?

**Comment by Lysxia** [comment]
Thanks for looking into this! Yes, I am happy with this PR.

**Comment by Alizter** [comment]
@coqbot merge now

**Review by Alizter — APPROVED** [review]

**Comment by Alizter** [comment]
@coqbot merge now

**Pull request merged**